src/Nethermind/Nethermind.Merge.Plugin/BlockProduction/IPayloadPreparationService.cs

```diff
@@ -12,5 +12,6 @@ public interface IPayloadPreparationService
         string? StartPreparingPayload(BlockHeader parentHeader, PayloadAttributes payloadAttributes);
 
         ValueTask<IBlockProductionContext?> GetPayload(string payloadId, bool skipCancel = false);
+        void CancelBlockProductionForParent(object? sender, BlockHeader e);
```

Comment:
Do we need this method or is it test only?
If yes it has weird/bad arguments

Reply:
Used for the simulate block production flag. Was not exposed via `IPayloadPreparationService`.

Reply:
Changed to `CancelBlockProduction(string payloadId)` and put the calling logic in `NewPayloadHandler`.